test/js/explore/scatter-tab.test.js

```diff
@@ -182,4 +185,81 @@ describe('getNewContextMap correctly assigns contexts', () => {
       expect(titleEl).toHaveTextContent(expectedTitles[index])
     })
   })
+
+  it('shows default group scatter plot and legend', async () => {
+    const apiFetch = jest.spyOn(ScpApi, 'fetchCluster')
+    // pass in a clone of the response since it may get modified by the cache operations
+    apiFetch.mockImplementation(params => {
+      const response = _cloneDeep(MANY_LABELS_MOCKS.CLUSTER_RESPONSE)
+      response.cluster = params.cluster
+      response.genes = params.genes
+      return Promise.resolve([response, CACHE_PERF_PARAMS])
+    })
+    const fakePlot = jest.spyOn(Plotly, 'react')
+    fakePlot.mockImplementation(() => {})
+    const fakeLogScatter = jest.spyOn(ScpApiMetrics, 'logScatterPlot')
+    fakeLogScatter.mockImplementation(() => {})
+
+
+    const { container } = render((
+      <ScatterTab studyAccession='SCP101'
+        exploreParams={{
+          'userSpecified': {
+            'annotation': true,
+            'cluster': true,
+            'spatialGroups': true,
+            'subsample': true
+          },
+          'cluster': 'cluster_many_long_odd_labels.tsv',
+          'annotation': {
+            'name': 'Category',
+            'type': 'group',
+            'scope': 'cluster'
+          },
+          'subsample': 'all',
+          'consensus': null,
+          'spatialGroups': [],
+          'genes': [],
+          'geneList': '',
+          'heatmapRowCentering': '',
+          'scatterColor': '',
+          'distributionPlot': '',
+          'distributionPoints': '',
+          'tab': '',
+          'heatmapFit': '',
+          'bamFileName': '',
+          'ideogramFileId': ''
+        }}
+        updateExploreParams={() => {}}
+        exploreInfo={MANY_LABELS_MOCKS.EXPLORE_RESPONSE}
+        isGene={false}
+        isMultiGene={false}
+        getPlotDimensions={() => [100, 100]}
+        dataCache={createCache()}/>
+    ))
+
+    // findByTestId would be more kosher, but the numerical leaf ("1") is
+    // assigned randomly in `ScatterPlot`.  This ID has been observed with the
+    // value "study-scatter-25", but relying on that seems brittle.
+    //
+    // Consider changing `ScatterPlot` to use a deterministic `data-testid`.
+    // await screen.findByTestId('study-scatter-1-legend')
+
+    await waitFor(() => {
+      container.querySelectorAll('#study-scatter-1-legend').length > 0
```

Comment:
Can this be `container.getElementById('#study-scatter-1-legend') != null`? If so, I think it's cleaner

Reply:
Surprisingly to me, it cannot: `TypeError: container.getElementById is not a function`.